### PR TITLE
takeWhileIncluding

### PR DESCRIPTION
**Pull request by davean**

### CODE REVIEW

File: Data/Attoparsec/ByteString/Internal.hs [hunk 1]
```diff
@@ -277,6 +278,47 @@ takeWhileAcc p = go
       else return $ concatReverse (s:acc)
 {-# INLINE takeWhileAcc #-}
 
+-- | Consume input until immediately after the predicate returns 'True', and return
+-- the consumed input.
+--
+-- This parser will consume at least one Word8 or fail.
```
Comment (bgamari) [inline Data/Attoparsec/ByteString/Internal.hs:284]:
```suggestion
-- This parser will consume at least one 'Word8' or fail.
```

File: Data/Attoparsec/ByteString/Internal.hs [hunk 2]
```diff
@@ -277,6 +278,47 @@ takeWhileAcc p = go
...
+      let s = s' `B8.snoc` h
+      advance (B8.length s)
+      pure s
+    -- The above isn't true so either we ran out of input or we need to process the next chunk.
+    Nothing -> do
+      continue <- inputSpansChunks (B8.length s')
+      if continue
+        then takeWhileIncAcc p [s']
+        -- Our spec says that if we run out of input we fail.
+        else fail "takeWhileIncluding"
```
Comment (bgamari) [inline Data/Attoparsec/ByteString/Internal.hs:301]: Perhaps fail with a more descriptive error?

File: Data/Attoparsec/ByteString/Internal.hs [hunk 3]
```diff
@@ -277,6 +278,47 @@ takeWhileAcc p = go
...
+     case B8.uncons t of
+       Just (h, _) -> do
+         let s = s' `B8.snoc` h
+         advance (B8.length s)
+         pure (concatReverse $ s:acc)
+       Nothing -> do
+         continue <- inputSpansChunks (B8.length s')
+         if continue
+           then go (s':acc)
+           else fail "takeWhileIncAcc"
```
Comment (bgamari) [inline Data/Attoparsec/ByteString/Internal.hs:318]: Same here.

**Review by bgamari — CHANGES_REQUESTED** [review]
Seems fine but a few nits inline.

File: Data/Attoparsec/ByteString/Internal.hs (hunk 2 above)
Comment (davean) [reply Data/Attoparsec/ByteString/Internal.hs]: I had wanted to match the library's standard and style, but ok, I've made it more verbose.

File: Data/Attoparsec/ByteString/Internal.hs (hunk 3 above)
Comment (davean) [reply Data/Attoparsec/ByteString/Internal.hs]: As above, same below.

**Comment by davean** [comment]
I had wanted the errors to match the style and standard of the library, but now there are more verbose messages.

**Comment by bgamari** [comment]
Looks good, thanks!

**Review by bgamari — APPROVED** [review]

**Pull request merged**